passageflex/src/main/java/id/passage/passageflex/PasskeyUtils.kt

```diff
@@ -0,0 +1,154 @@
+package id.passage.passageflex
+
+import android.app.Activity
+import androidx.credentials.CreateCredentialResponse
+import androidx.credentials.CreatePublicKeyCredentialRequest
+import androidx.credentials.CredentialManager
+import androidx.credentials.GetCredentialRequest
+import androidx.credentials.GetCredentialResponse
+import androidx.credentials.GetPublicKeyCredentialOption
+import androidx.credentials.exceptions.CreateCredentialException
+import androidx.credentials.exceptions.GetCredentialException
+import com.squareup.moshi.Moshi
+import id.passage.android.passageflex.model.CredentialAssertionChallenge
+import id.passage.android.passageflex.model.CredentialAssertionResponse
+import id.passage.android.passageflex.model.CredentialAssertionResponseJsonAdapter
+import id.passage.android.passageflex.model.CredentialCreationChallenge
+import id.passage.android.passageflex.model.CredentialCreationPublicKeyJsonAdapter
+import id.passage.android.passageflex.model.CredentialCreationResponse1
+import id.passage.android.passageflex.model.CredentialCreationResponse1JsonAdapter
+import id.passage.android.passageflex.model.ProtocolCredentialAssertionPublicKey
+import id.passage.android.passageflex.model.ProtocolCredentialAssertionPublicKeyJsonAdapter
+import id.passage.passageflex.CredentialParsingException.Companion.CHALLENGE_MISSING
+import id.passage.passageflex.CredentialParsingException.Companion.CHALLENGE_PARSING_FAILED
+import id.passage.passageflex.CredentialParsingException.Companion.CREDENTIAL_PARSING_FAILED
+
+@Suppress("unused", "RedundantVisibilityModifier", "RedundantModalityModifier")
+public object PasskeyUtils {
+    private const val REGISTRATION_RESPONSE_BUNDLE_KEY = "androidx.credentials.BUNDLE_KEY_REGISTRATION_RESPONSE_JSON"
+    private const val AUTH_RESPONSE_BUNDLE_KEY = "androidx.credentials.BUNDLE_KEY_AUTHENTICATION_RESPONSE_JSON"
+
+    /**
+     * Registers a user passkey credential that can be used to authenticate the user to
+     * the app in the future.
+     *
+     * The execution potentially launches framework UI flows for a user to view their registration
+     * options, grant consent, etc.
+     *
+     * @param requestJson The privileged request in JSON format in the [standard webauthn web json](https://w3c.github.io/webauthn/#dictdef-publickeycredentialcreationoptionsjson).
+     * @throws CreateCredentialException If the request fails
+     */
+    public suspend fun createPasskey(
```

Comment:
Another possibly heavily web FE biased (and my own personal bias haha) but I wonder what your thoughts are on separating out one util per file. There's nothing wrong with having a single util file and certainly nothing wrong with the way you have this laid out as the functions are in a similar vein, but I have seen these utils file sometimes turn into catch-alls over time that lead to code smells.

Again - my personal preference, a nitpick, and might not even make sense for mobile dev, I'm just curious!

Reply:
Thanks for the input! I may break up this class in the future, but think its okay for now :)